components/console-backend-service/internal/domain/servicecatalog/clusterserviceplan_resolver.go

```diff
@@ -0,0 +1,44 @@
+package servicecatalog
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/golang/glog"
+	"github.com/kyma-project/kyma/components/console-backend-service/internal/domain/shared"
+	"github.com/kyma-project/kyma/components/console-backend-service/internal/gqlerror"
+	"github.com/kyma-project/kyma/components/console-backend-service/internal/gqlschema"
+)
+
+type clusterServicePlanResolver struct {
+	sc     *clusterServicePlanService
+	rafter shared.RafterRetriever
+}
+
+func newClusterServicePlanResolver(sc *clusterServicePlanService, r shared.RafterRetriever) *clusterServicePlanResolver {
+	return &clusterServicePlanResolver{sc, r}
+}
+
+func (r *clusterServicePlanResolver) ClusterServicePlanClusterAssetGroupField(ctx context.Context, obj *gqlschema.ClusterServicePlan) (*gqlschema.ClusterAssetGroup, error) {
+	if obj == nil {
+		glog.Error(fmt.Errorf("while getting clusterAssetGroup field obj is empty"))
+		return nil, gqlerror.NewInternal()
+	}
+	return r.getClusterAssetGroup(obj.Name), nil
+}
+
+func (r *clusterServicePlanResolver) getClusterAssetGroup(name string) *gqlschema.ClusterAssetGroup {
```

Comment:
This function also is not needed. You can add whole logic only in one function `ClusterServicePlanClusterAssetGroupField`, not split to two.